src/components/TodosFilter.tsx

```diff
@@ -0,0 +1,80 @@
+import { useContext } from 'react';
+import { TodosContext } from './TodosContext';
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+
+export const TodosFilter: React.FC = () => {
+  const {
+    todos,
+    handleStatus,
+    selectedStatus,
+    setTodos,
+  } = useContext(TodosContext);
+
+  const ItemsLeft = () => {
+    return todos.filter(todo => todo.completed === false).length;
+  };
+
+  const handleDeleteClick = () => {
```

Comment:
The `handleDeleteClick` function is redefined on each render. Use `useCallback` hook to prevent unnecessary function redefinition.